src/main/scala/is/hail/methods/LogisticRegression.scala

```diff
@@ -61,8 +65,14 @@ object LogisticRegression {
     vds.copy(rdd = vds.rdd.mapPartitions( { it =>
       val X = XBc.value.copy
       it.map { case (v, (va, gs)) =>
+
         val logregAnnot =
-          if (RegressionUtils.setLastColumnToMaskedGts(X, gs.hardCallGenotypeIterator, sampleMaskBc.value))
+          if ((!useDosages &&
+            RegressionUtils.setLastColumnToMaskedGts(X, gs.hardCallGenotypeIterator,
```

Comment:
This is confusing code. Why does something called `setLastColumn...` return a boolean? Is there a side effect?